01/app.js

```diff
@@ -1,21 +1,35 @@
-document.addEventListener('DOMContentLoaded', init);
 
-function init() {
-    const divList = document.querySelectorAll('div');
-    
-    setBorderColorAsync(divList[0], 'red', function() {
-        setBorderColorAsync(divList[1], 'blue', function() {
-            setBorderColorAsync(divList[2], 'green', function() {
-                console.log('finish');
-            });
-        });
-    });
+const divList = document.querySelectorAll('div');
+
+const sayStart = () => {
+    return new Promise((resolve,reject) => {
+        setTimeout(() => {
+            console.log(`start`);
+            resolve()
+        },1000)
+    })
+}
 
+const setBorderColorAsync = (element,color,time) => {
+    return new Promise ((resolve, reject) => {
+        setTimeout(() => {
+            element.style.border = `3px solid ${color}`;
+            resolve();
+        },time)
+    });
 }
 
-function setBorderColorAsync(element, color, callback) {
-    setTimeout(() => {
-        element.style.border = `3px solid ${color}`;
-        callback();
-    }, Math.random() * 3000);
-}
\ No newline at end of file
+let getTime = () => { return time = Math.random() * 3000;}
+
+sayStart()
+    .then(() => {
+        return setBorderColorAsync(divList[0],'red', getTime())
+    })
+    .then(() => {
+        return setBorderColorAsync(divList[1],'blue', getTime())
+    })
+    .then(() => {
+        return setBorderColorAsync(divList[2],'green', getTime())
+    })
+    .then(() => console.log('end'))
+
```

Comment:
Zgadza się, tutaj bardziej chodziło o to, aby stworzyć funkcje z nazwę i w odpowiednim miejscu ją przekazywać/uruchamiać ;) Może zostać - ta opcja jest trudniejsza ;P